packages/@ember/-internals/metal/lib/computed_cache.ts

```diff
@@ -1,59 +1,69 @@
-const COMPUTED_PROPERTY_LAST_REVISION = new WeakMap<object, Map<string, number>>();
+import { Meta } from '@ember/-internals/meta';
 
-export function getCacheFor(obj: object): Map<string, any> {
-  let cache = COMPUTED_PROPERTY_CACHED_VALUES.get(obj);
-  if (cache === undefined) {
-    cache = new Map<string, any>();
+// const COMPUTED_PROPERTY_CACHED_VALUES = new WeakMap<object, Map<string, any | null | undefined>>();
+// const COMPUTED_PROPERTY_LAST_REVISION = new WeakMap<object, Map<string, number>>();
 
-    COMPUTED_PROPERTY_CACHED_VALUES.set(obj, cache);
-  }
-  return cache;
+export function valueFor(meta: Meta, key: string) {
+  return meta.valueFor(key);
 }
 
-/**
-  Returns the cached value for a property, if one exists.
-  This can be useful for peeking at the value of a computed
-  property that is generated lazily, without accidentally causing
-  it to be created.
-
-  @method cacheFor
-  @static
-  @for @ember/object/internals
-  @param {Object} obj the object whose property you want to check
-  @param {String} key the name of the property whose cached value you want
-    to return
-  @return {Object} the cached value
-  @public
-*/
-export function getCachedValueFor(obj: object, key: string): any {
-  let cache = COMPUTED_PROPERTY_CACHED_VALUES.get(obj);
-  if (cache !== undefined) {
-    return cache.get(key);
-  }
+export function setValueFor(obj: object, key: string, value: unknown, meta: Meta) {
+  meta.setValueFor(key, value);
 }
 
-export function setLastRevisionFor(obj: object, key: string, revision: number): void {
-  let cache = COMPUTED_PROPERTY_LAST_REVISION!.get(obj);
+// export function getCacheFor(obj: object, ): Map<string, any> {
+//   let cache = COMPUTED_PROPERTY_CACHED_VALUES.get(obj);
+//   if (cache === undefined) {
+//     cache = new Map<string, any>();
 
-  if (cache === undefined) {
-    cache = new Map<string, any>();
-    COMPUTED_PROPERTY_LAST_REVISION!.set(obj, cache);
-  }
+//     COMPUTED_PROPERTY_CACHED_VALUES.set(obj, cache);
+//   }
+//   return cache;
+// }
 
-  cache!.set(key, revision);
-}
+// /**
+//   Returns the cached value for a property, if one exists.
+//   This can be useful for peeking at the value of a computed
+//   property that is generated lazily, without accidentally causing
+//   it to be created.
 
-export function getLastRevisionFor(obj: object, key: string): number {
-  let cache = COMPUTED_PROPERTY_LAST_REVISION!.get(obj);
-  if (cache === undefined) {
-    return 0;
-  } else {
-    let revision = cache.get(key);
-    return revision === undefined ? 0 : revision;
-  }
-}
+//   @method cacheFor
+//   @static
+//   @for @ember/object/internals
+//   @param {Object} obj the object whose property you want to check
+//   @param {String} key the name of the property whose cached value you want
+//     to return
+//   @return {Object} the cached value
+//   @public
+// */
+// export function getCachedValueFor(obj: object, key: string): any {
```

Comment:
did you mean to comment out or delete?

Reply:
meant to delete, thanks for catching that